brvs-core/src/main/java/iroha/validation/rest/RestService.java

```diff
@@ -77,11 +80,31 @@ public Response register(@PathParam("accountId") String accountId) {
   @Consumes(MediaType.APPLICATION_JSON)
   @Produces(MediaType.APPLICATION_JSON)
   public Response sendTransaction(String transaction) {
+    return sendTransaction(transaction, false);
+  }
+
+  @POST
+  @Path("/transaction/sign")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response sendTransactionWithSigning(String transaction) {
+    return sendTransaction(transaction, true);
+  }
+
+  private Response sendTransaction(String transaction, boolean sign) {
     try {
       final Builder builder = Transaction.newBuilder();
       parser.merge(transaction, builder);
-      final Transaction tx = builder.build();
-      StreamingOutput streamingOutput = output -> irohaAPI.transaction(tx)
+      // since final or effectively final is needed
+      final Transaction builtTx = builder.build();
+      final Transaction transactionsToSend;
+      if (sign) {
+        transactionsToSend = jp.co.soramitsu.iroha.java.Transaction.parseFrom(builtTx)
+            .sign(brvsAccountKeyPair).build();
+      } else {
+        transactionsToSend = builtTx;
+      }
+      StreamingOutput streamingOutput = output -> irohaAPI.transaction(transactionsToSend)
           .subscribeOn(Schedulers.from(Executors.newSingleThreadExecutor()))
```

Comment:
Can we just reuse the same thread pool rather than creating it on every call? And who shuts the thread pool down?

Reply:
I just wanted to isolate all calls completely so I create a new one every time